src/WebJobs.Script/Extensions/AssemblyExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public static class AssemblyExtensions
+    {
+        private readonly static ConcurrentDictionary<Assembly, string> _codeBaseMapping = new ConcurrentDictionary<Assembly, string>();
+
+        public static bool MapCodeBase(this Assembly assembly, string codeBasePath)
+        {
+            var uri = new Uri(codeBasePath);
+            return _codeBaseMapping.TryAdd(assembly, uri.AbsoluteUri);
```

Comment:
Are there cases where someone might have loaded the assembly from one location, then say changed their directory structure and load it from another location? This would through an exception in this case.